### ISSUE TITLE
position of nn.SpatialBatchNormalization

### ISSUE THREAD

**Issue by zhfe99** [eladhoffer/ImageNet-Training#14]
In https://github.com/eladhoffer/ImageNet-Training/blob/master/Models/AlexNet_BN.lua, each `nn.SpatialBatchNormalization` is just after the nonlinear `cudnn.ReLU` and `nn.Threshold`. 

But in the original batch normalization paper, the authors suggest to apply Batch Normalization to the input of each non-linearity. In https://github.com/soumith/imagenet-multiGPU.torch/blob/master/models/alexnetowtbn_cudnn.lua, it also adds the Batch Normalization before `cudnn.ReLU`and `nn.Threshold`. 

I am not sure whether this difference could decrease accuracy?

**Comment by eladhoffer** [comment]
It doesn't make a difference...
